### PR TITLE
Fix: matching a lambda expression with a functional-interface that wasn't annotated with @FunctionalInteface didn't work.

### PR DESCRIPTION
**Pull request by maartenc**

### CODE REVIEW

**Comment by matozoid** [comment]
Thanks! Let's see what @ftomassetti says.

File: javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/model/typesystem/ReferenceTypeImpl.java [hunk 1]
```diff
@@ -106,7 +107,7 @@ public boolean isAssignableBy(ResolvedType other) {
             }
         }
         if (other instanceof LambdaArgumentTypePlaceholder) {
-            return this.getTypeDeclaration().hasAnnotation(FunctionalInterface.class.getCanonicalName());
+            return FunctionalInterfaceLogic.isFunctionalInterfaceType(this);
```
Comment (ftomassetti) [inline javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/model/typesystem/ReferenceTypeImpl.java:110]: Seems better

**Review by ftomassetti — APPROVED** [review]
Seems good to me

**Comment by ftomassetti** [comment]
Update code, let's see if tests still pass

**Pull request merged**